### PR TITLE
Add subcommand to update repositories

### LINKED ISSUE #22: Pull changes for any project from anywhere
**Is your feature request related to a problem? Please describe.**
Since we are using devmode and not github to manage the project using devmode for simple tasks from everywhere with out the need to open the project from the specific folder, one of the cool features of devmode is that is a manager of projects from everywhere on the deskt and on any popular system [ Mac / Linux / Windows ].

**Describe the solution you'd like**
I would like to add a command in wich you can upgrade or retrive information from a project using only the command dm update [project]

**Describe alternatives you've considered**
None at all 

**Additional context**
Since devmode is a tool to have super powers all over the system to open and manage your projects why not do it in the right way?

### PR DESCRIPTION
**Pull request by Kedap**
fix #22

### CODE REVIEW

File: src/cli.rs [hunk 1]
```diff
@@ -337,7 +351,7 @@ impl Cli {
                         Colorize::yellow(&*name),
                         Colorize::blue(&*rename.unwrap())
                     );
-                } else if add.is_some() {
+                } else if let Some(..) = add {
```
Comment (edfloreshz) [inline src/cli.rs:354]: If you don't need the `T` inside an `Option<T>`, it is recommended to use `is_some()`.

```suggestion
                } else if add.is_some() {
```

File: src/cli.rs [hunk 2]
```diff
@@ -366,13 +378,13 @@ impl Cli {
...
-                                move_items(&vec![path.clone()], to, &options)?;
+                                move_items(&[path], to, &options)?;
                             }
                         }
                     } else {
-                        move_items(&vec![path.clone()], to, &options)?;
+                        move_items(&[path], to, &options)?;
                     }
-                } else if remove.is_some() {
+                } else if let Some(..) = remove {
```
Comment (edfloreshz) [inline src/cli.rs:387]: Same here.
```suggestion
                } else if remove.is_some() {
```

File: src/cli.rs [hunk 3]
```diff
@@ -58,6 +58,16 @@ pub enum Commands {
         #[clap(takes_value = true, required = true)]
         project: String,
     },
+    #[clap(
+        about = "Updates a project.",
```
Comment (edfloreshz) [inline src/cli.rs:62]:
```suggestion
        about = "Fetch the latest changes for a project.",
```

**Review by edfloreshz — CHANGES_REQUESTED** [review]
Please review some of the comments before we merge this, good job! 👍🏼

**Comment by Kedap** [comment]
It seems that it was my mistake for not verifying well what [clippy said](https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_unwrap) and not completely refactoring

**Comment by edfloreshz** [comment]
No worries, you can do `cargo clippy--fix` to apply all the suggestions automatically, be sure to commit your changes before, since this might be a destructive action.

**Comment by edfloreshz** [comment]
I tried to use `update` but got the following result:
image
I had to pull the changes manually.

File: src/utils/project.rs [hunk 4]
```diff
@@ -102,6 +130,36 @@ pub fn open_project(name: &str, paths: Vec<String>) -> Result<()> {
...
+        "main"
+    };
+    project
+        .find_remote("origin")?
+        .fetch(&[main_branch], None, None)?;
+    let fetch_head = project.find_reference("FETCH_HEAD")?;
+    let fetch_commit = project.reference_to_annotated_commit(&fetch_head)?;
+    let analysis = project.merge_analysis(&[&fetch_commit])?;
+
+    if analysis.0.is_fast_forward() {
```
Comment (edfloreshz) [inline src/utils/project.rs:151]: The error should be somewhere here.

**Review by edfloreshz — CHANGES_REQUESTED** [review]
Please let me know if you need any assistance.

File: src/utils/project.rs [hunk 5]
```diff
@@ -133,31 +132,9 @@ pub fn open_project(name: &str, paths: Vec<String>) -> Result<()> {
 pub fn update_project(name: &str, paths: Vec<String>) -> Result<()> {
     println!("Update project {}... \n\n", name);
```
Comment (edfloreshz) [inline src/utils/project.rs:133]:
```suggestion
    println!("Updating project {}... \n\n", name);
```

File: src/utils/project.rs [hunk 6]
```diff
@@ -133,31 +132,9 @@ pub fn open_project(name: &str, paths: Vec<String>) -> Result<()> {
...
     let path = &paths[0];
-    let project = Repository::open(Path::new(path)).expect(NO_PROJECT_FOUND);
     // project.reset(&project.revparse_single("HEAD"), ResetType::Hard, None)?;
```
Comment (edfloreshz) [inline src/utils/project.rs:135]:
```suggestion

```

File: src/utils/git_pull.rs [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+use std::io;
+use std::io::Write;
+use std::path::Path;
+
+use anyhow::bail;
+use anyhow::Result;
+
+use crate::constants::messages::*;
+
+pub fn pull(repo_paht: &Path) -> Result<()> {
```
Comment (edfloreshz) [inline src/utils/git_pull.rs:12]:
```suggestion
pub fn pull(repo_path: &Path) -> Result<()> {
```

File: src/utils/git_pull.rs [hunk 8]
```diff
@@ -0,0 +1,184 @@
+use git2::ErrorCode;
+use git2::Repository;
+use std::io;
+use std::io::Write;
+use std::path::Path;
+
+use anyhow::bail;
+use anyhow::Result;
```
Comment (edfloreshz) [inline src/utils/git_pull.rs:8]:
```suggestion
use anyhow::{Result, Context};
```

**Review by edfloreshz — CHANGES_REQUESTED** [review]
Looking good, just a few changes and we're done!

File: src/utils/git_pull.rs [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+
+use crate::constants::messages::*;
+
+pub fn pull(repo_paht: &Path) -> Result<()> {
+    let remote_name = "origin";
+    let repo = Repository::open(repo_paht).expect(NO_PROJECT_FOUND);
```
Comment (edfloreshz) [inline src/utils/git_pull.rs:14]:
```suggestion
    let repo = Repository::open(repo_path).with_context(|| NO_PROJECT_FOUND);
```

**Comment by edfloreshz** [comment]
Thank you!

**Pull request merged**